src/backend/distributed/executor/multi_task_tracker_executor.c

```diff
@@ -3004,3 +3007,37 @@ TrackerHashDisconnect(HTAB *taskTrackerHash)
 		taskTracker = (TaskTracker *) hash_seq_search(&status);
 	}
 }
+
+
+/*
+ * TaskTrackerExecScan is a callback function which returns next tuple from a
+ * task-tracker execution. In the first call, it executes distributed task-tracker
+ * plan and loads results from temporary files into custom scan's tuple store.
+ * Then, it returns tuples one by one from this tuple store.
+ */
+TupleTableSlot *
+TaskTrackerExecScan(CustomScanState *node)
+{
+	CitusScanState *scanState = (CitusScanState *) node;
+	TupleTableSlot *resultSlot = NULL;
+
+	if (!scanState->finishedRemoteScan)
+	{
+		DistributedPlan *distributedPlan = scanState->distributedPlan;
+		Job *workerJob = distributedPlan->workerJob;
+
+		/* we are taking locks on partitions of partitioned tables */
+		LockPartitionsInRelationList(distributedPlan->relationIdList, AccessShareLock);
```

Comment:
I've been wondering, why isn't this in `CitusSelectBeginScan`?

Reply:
I saw it in your PR, so didn't want to interfere with your changes. Also, since this commit only moves files rather than modifying, I'd like to keep it as it is.